dashboard-app/src/lib/ui/components/Boundaries/ErrorBoundary.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { AxiosError } from 'axios';
+import { propOr } from 'ramda';
+
+
+type ErrorBoundaryState = {
+  hasError: boolean
+  isHttpError: boolean
+  statusCode: number
+  message: string
+};
+
+const isHttpError = (error: Error) =>
+  ['isAxiosError', 'response']
+    .reduce((acc, s) => acc || error.hasOwnProperty(s), false);
+
+class Boundary<T extends {}> extends React.Component<T, ErrorBoundaryState> {
+  constructor (props: Readonly<T>) {
+    super(props);
+    this.state = {
+      hasError: false,
+      isHttpError: false,
+      statusCode: 0,
+      message: '',
+    };
+  }
+
+  static getDerivedStateFromError (error: Error | AxiosError) {
+    return {
+      hasError: true,
+      isHttpError: isHttpError(error),
+      statusCode: propOr(0, 'code', error),
+      message: error.message,
+    };
+  }
+
+  componentDidCatch (error: Error, errorInfo: { componentStack: any }) {
+    if (process.env.NODE_ENV !== 'test') {
+      console.log(error, errorInfo);
+    }
```

Comment:
Would be rad if we could sent the error to the server here. I'm leaning to think that these boundaries should be added as a separate PR so we can do styling n server changes in one.

Reply:
What should we do if there's an error? We could fall back to using the placeholders, but if there's an error in fetching the data on the dashboard there's probably gonna be one on the rest of the dashboard too, so not sure it makes sense to let them continue.

What I should probably do is create two error boundaries, one which renders something itself, and one which redirects (probs by rendering a `<Redirect>` component), and then we can do the same redirect behaviour that we currently have on API calls.

The former component can be used for failures in non-critical components where it's ok for us to just render some placeholder without the app becoming unusable.